### ISSUE TITLE
allow changing of ageClassCutoffs?

### ISSUE THREAD

**Issue by achubaty** [PredictiveEcology/LandWeb#86]
Age classes were decided by consensus and standardized so that we can make comparisons among different areas. However, some partners want to be able to use their own cutoffs for reporting.

How feasible is this? Is it simply an app/reporting change, or will this affect the underlying model? We can't rerun custom models for partners -- they should be using the app.

Currently:

```r
ageClassCutOffs <- c(0, 40, 80, 120)
```

**Comment by achubaty** [comment]
**Scoping update (2026-08-06).** The ask is a mechanism to pass age-class cutoffs **per species**, via a table, so partners can report against their own cutoffs. Proposed shape: a **long `data.frame`** with a `zone` / `type` column keyed to FMA or jurisdiction, plus species and cutoff bounds, holding every variant in one place — including a clearly-defined **`"default"`** entry carrying today's uniform values (`c(0L, 40L, 80L, 120L)` / `Young`, `Immature`, `Mature`, `Old`). Every run must report **both**: custom cutoffs for the partner's FMP reporting, and uniform defaults for cross-area comparison in LandWeb reports.

**Confirmed: this is reporting-only.** `ageClassCutOffs` never reaches the simulation — it appears only in `NRV_summary` (params, the metrics call, and the legend), `HSI_Caribou_MB`, and the analysis packages `LandWebUtils` (`leading.R`, `largePatches.R`, `misc.R`) and [`nrvtools`](https://github.com/FOR-CAST/nrvtools) (`landweb-summaries.R`). Nothing in `Biomass_*`, `LandMine`, `burnSummaries`, or `_targets.R`. The cutoffs are applied post-hoc to the saved per-rep `vtm` + `age` rasters, so **both** cutoff sets can be produced from the same rasters **with no re-run** — which is what makes the dual custom+default requirement cheap.

**Decided: fixed 4 classes, varying bounds.** `Young`/`Immature`/`Mature`/`Old` stays fixed; only the numeric bounds vary by species and zone. Variable class counts are out of scope. Consequently `nrvtools:::.age_class_raster()`'s `length(cutoffs) == length(classes)` invariant holds unchanged, the palette / facets / column schemas stay fixed, `ageClasses` remains a package constant, and the table is exactly **4 rows per (`type` × species)** — validation reduces to: 4 rows per group, bounds strictly increasing, first bound `0`.

**Per-species is a deeper change than per-zone.** Per-zone is a lookup swap; per-species changes the algorithm. `.age_class_raster()` currently classifies the age raster *once, uniformly*, and `leadingVegByAgeClass()` then cross-tabulates against the leading-species map — per-species bounds make that classification conditional on the leading species at each pixel. `LandWebUtils::LargePatches()` uses the same reclass, so Leading **and** LargePatches both change.

**Knock-on effects to settle during design:**

1. **Legends can only show class names** — no single numeric range exists, so bounds move to captions or a per-species table.
2. **The `"All species"` roll-up becomes ill-defined** — if `Mature` means different ages for different species it aggregates unlike things. Either drop it under custom cutoffs, or define it explicitly (e.g. always roll up on the default cutoffs) and label it.
3. **Cross-area / cross-species comparison breaks** — fixed labels, varying bounds. This is why the uniform-default output must *always* accompany the custom one.
4. **The current-conditions overlay must use matching cutoffs** — a custom-cutoff envelope with a default-cutoff `_CC` reference is an invalid comparison, so CC gets computed twice too.
5. **Outputs need a cutoff-set discriminator** — `_aggregates/<refCode>/`, envelope CSVs, and figure names are keyed only on the reporting polygon, so without one the two sets silently overwrite each other. ⚠️ most likely way to get this wrong.
6. **Postprocessing roughly doubles** for Leading + LargePatches — post-hoc raster work, not a re-run, but compute both sets in one pass.
7. **Reports must label which set each figure uses** — the class names are identical either way, so a reader cannot infer it.

**Landing zone:** the mechanism belongs in `nrvtools` and must stay **class-count-agnostic**, with LandWeb's 4-class table in `LandWebUtils`. Note **BC_HRV** must adopt the same mechanism and already runs an **8-class** scheme (`Young1`…`Old2`, `seq(0, 140, 20)`), while separately hardcoding a verbatim copy of the LandWeb defaults in `SpaDES/R/animation_tsf.R` — so baking "4 classes" into `nrvtools` would break it. A BC_HRV issue should follow once this lands.

Keeping this issue here rather than moving it to the app repo: the app only needs to select which pre-generated set to display.

**Comment by achubaty** [comment]
Cross-project follow-up filed: FOR-CAST/BC_HRV#6 — BC_HRV must adopt this mechanism once it lands. Note BC already runs an **8-class** scheme, so the shared `nrvtools` layer must stay class-count-agnostic; only LandWeb's *table* fixes on 4 classes.
